core/src/main/java/tech/pegasys/web3signer/core/service/jsonrpc/handlers/signing/TransactionSerializer.java

```diff
@@ -73,11 +76,12 @@ private static byte[] prependEip1559TransactionType(byte[] bytesToSign) {
         .array();
   }
 
-  private SignatureData sign(final byte[] bytesToSign) {
-    final SecpArtifactSignature artifactSignature =
-        (SecpArtifactSignature) signer.sign(Bytes.of(bytesToSign));
+  private SignatureData sign(final String eth1Address, final byte[] bytesToSign) {
+    final String bytesSigned =
+        secpSigner.sign(normaliseIdentifier(eth1Address), Bytes.of(bytesToSign)).orElseThrow();
 
-    final Signature signature = artifactSignature.getSignatureData();
+    final Signature signature =
```

Comment:
nit: not ideal going from Bytes -> String -> byte[] -> SecpArtifactSignature. Do we need a sign method on SignerForIdentifier that will return Bytes instead?